### PR TITLE
docker: reduce size of image after building

### PR DESCRIPTION
**Pull request by dungwinux**
- [x] reduced from 19.8GB -> 14.8GB according to `docker images`
- [x] Re-run all benchmarks to make sure nothing breaks

### CODE REVIEW

File: docker/main.Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ RUN apt update
 RUN DEBIAN_FRONTEND=noninteractive apt install -y tzdata
 RUN apt install -y git cmake ninja-build sudo curl wget pkg-config gnupg
 
-COPY docker /root/Energy-Languages/docker
+COPY docker/keys /root/Energy-Languages/docker/keys
```
Comment (nicovank) [inline docker/main.Dockerfile:10]: Right, I changed this by accident.

File: docker/main.Dockerfile [hunk 2]
```diff
@@ -25,49 +25,49 @@ RUN apt install -y libapr1-dev libgmp-dev libpcre3-dev libboost-regex-dev
...
-RUN gpg --verify rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz.asc rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz
-RUN tar -xzf rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz
-RUN ./rust-${RUST_VERSION}-x86_64-unknown-linux-gnu/install.sh
-RUN rm rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz.asc
+RUN wget --no-verbose https://static.rust-lang.org/dist/rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz \
+ && wget --no-verbose https://static.rust-lang.org/dist/rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz.asc \
+ && gpg --verify rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz.asc rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz \
+ && tar -xzf rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz \
+ && ./rust-${RUST_VERSION}-x86_64-unknown-linux-gnu/install.sh \
+ && rm -rf rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz rust-${RUST_VERSION}-x86_64-unknown-linux-gnu.tar.gz.asc rust-${RUST_VERSION}-x86_64-unknown-linux-gnu
```
Comment (nicovank) [inline docker/main.Dockerfile]: Nit, can you have 4 spaces between start of line and `&&`?

Comment (dungwinux) [reply docker/main.Dockerfile]: I fixed it in a new commit

**Comment by nicovank** [comment]
`14.8GB` I wonder what's still taking space. Did you ever have any time to check layer by layer?

**Comment by dungwinux** [comment]
I was doing this manually; the `dive` tool actually extracts the image and loads all of it in memory, which is certainly impossible for 16GB of RAM. I can try again with the new image

**Comment by nicovank** [comment]
Let me try on server, it has enough RAM. I have some free time today

**Comment by nicovank** [comment]
`fasta-800000000.txt` is ~8GB so that explains a lot.... It was needed for fixed-time experiment which we aren't using in the project anymore. At some point I'll delete it all.

**Review by nicovank — APPROVED** [review]

**Comment by nicovank** [comment]
FYI If you just want layer sizes, seems like this works also:  `docker history --no-trunc energy-languages`.
`dive` has some layer scrolling issue when I tried it, couldn't display > ~10 layers. When I searched GitHub it looks like I'd have to rebuilding from another branch to fix it.

**Comment by nicovank** [comment]
Just removed `Fixed-Time` experiment. Merge this, then try again, and let me know what the size is.

**Comment by nicovank** [comment]
```
# Old image.
energy-languages   latest    8a0ef03b2cb1   26 hours ago   18.1GB

# New image.
energy-languages   latest    945129cb5c7d   8 seconds ago   6.65GB
```

**Pull request merged**